data_storage.cpp

```diff
@@ -0,0 +1,106 @@
+#include "data_storage.h"
+
+
+/*
+* std::string filename	-- ��� �����-��������� ������� ������ �� �����
+* size_t block_size		-- ������ ����� ������ (� ������)
+*/
+DataStorage::DataStorage(std::string filename, size_t block_size, bool reinitialize)
+	: _filename(filename)
+	, _block_size(block_size) {
+	if (reinitialize) {
+		data_storage_stream = std::move(std::fstream (_filename, std::ios::in | std::ios::out | std::ios::binary | std::ios::trunc));
+	}
```

Comment:
auto mode = std::ios::in | std::ios::out | std::ios::binary
if(reinitialize)
mode  = mode | std::ios::trunc;
data_storage_stream = std::move(std::fstream (_filename, mode));